codex-rs/mcp-server/src/codex_message_processor.rs

```diff
@@ -669,6 +674,81 @@ impl CodexMessageProcessor {
         }
     }
 
+    async fn handle_delete_conversation(
+        &self,
+        request_id: RequestId,
+        params: DeleteConversationParams,
+    ) {
+        let DeleteConversationParams {
+            conversation_id,
+            rollout_path,
+        } = params;
+
+        let conversation = match self
+            .conversation_manager
+            .get_conversation(conversation_id)
+            .await
+        {
+            Ok(conversation) => conversation,
+            Err(_) => {
+                if let Some(path) = rollout_path {
+                    if let Err(err) = RolloutRecorder::delete_rollout_file_at_path(&path).await {
+                        let error = JSONRPCErrorError {
+                            code: INTERNAL_ERROR_CODE,
+                            message: format!(
+                                "failed to delete rollout file {}: {err}",
+                                path.display()
+                            ),
+                            data: None,
+                        };
+                        self.outgoing.send_error(request_id, error).await;
+                        return;
+                    }
+
+                    self.outgoing
+                        .send_response(request_id, DeleteConversationResponse {})
+                        .await;
+                } else {
+                    let error = JSONRPCErrorError {
+                        code: INVALID_REQUEST_ERROR_CODE,
+                        message: format!(
+                            "conversation {conversation_id} not found and no rollout path provided"
+                        ),
+                        data: None,
+                    };
+                    self.outgoing.send_error(request_id, error).await;
+                }
+                return;
+            }
+        };
+        if let Err(err) = conversation
+            .submit(Op::Shutdown {
+                delete_rollout: true,
```

Comment:
The only time this is set to true is if you get an error?

I don't understand why we would ever delete the rollout file. We only delete conversations to free up memory.